test-helpers2.js

```diff
@@ -0,0 +1,407 @@
+window.injectTest = (function() {
+
+'use strict';
+
+angular
+.module('ngTest', [])
+.service('Test', Test);
+
+function Test() {
+  this.Serv = Serv;
+  this.Element = Element;
+  this.Module = Module;
+
+  var core = {};
+
+  inject(CoreServices);
+  function CoreServices($rootScope, $compile, $timeout, $controller, $httpBackend, $templateCache, $filter,
+    $injector, $componentController) {
+    angular.extend(core, {
+      $rootScope: $rootScope,
+      $compile: $compile,
+      $timeout: $timeout,
+      $controller: $controller,
+      $httpBackend: $httpBackend,
+      $templateCache: $templateCache,
+      $filter: $filter,
+      $injector: $injector,
+      $componentController: $componentController
+    });
+  }
+
+  function Serv(name) {
+    if (name) {
+      return core.$injector.get(name);
+    }
+  }
+
+  // This method allows you to add any number of methods as a list of string method names which return promises like so:
+  // myMockedService.addPromise('methodA', 'methodB', 'methodC');
```

Comment:
If you are adding comments here, you should use JSDoc convention: http://usejsdoc.org/about-getting-started.html